lms/djangoapps/courseware/user_state_client.py

```diff
@@ -425,7 +426,10 @@ def iter_all_for_block(self, block_key, scope=Scope.user_state):
                 if state == {}:
                     continue
 
-                yield XBlockUserState(sm.student.username, sm.module_state_key, state, sm.modified, scope)
+                try:
+                    yield XBlockUserState(sm.student.username, sm.module_state_key, state, sm.modified, scope)
+                except ObjectDoesNotExist:
+                    log.warning("User Not Available: {0}".format(sm.student_id))
```

Comment:
```python
log.warning('The student with id %s from student module does not exist', sm.student_id)
```

Also add a new block for try/catch on `sm.student`

Reply:
Done